public/js/eventsPage.js

```diff
@@ -208,24 +210,59 @@ async function displayHiringPartnersAttending(event){
  *
  * @param events an object which contains information about an event
  */
-async function eventGenerator(event, hiringPartners) {
+async function eventGenerator(event, hiringPartners, applicants) {
     let eventInformation = ''
     let date = new Date(event.date).toDateString()
+    let numberOfAttendees
+    if (event.category_name === 'Assessment') {
+        let assessmentApplicants = applicants.filter(function (applicant) {
+            if (applicant.assessmentDay == event.id) {
+                return applicant
+            }
+            })
```

Comment:
This works better. Because filter only needs to return a boolean value (if true, it keeps the item, if false it removes it) you could simply the logic here like this:
```
let assessmentApplicants = applicants.filter(function (applicant) {
    return applicant.assessmentDay === event.id
})
```

If you used a fat arrow function you could make it even more compact:

```
let assessmentApplicants = applicants.filter(applicant => applicant.assessmentDay === event.id)
```

Reply:
That is much prettier! JS seems like so long ago now :'(